lua/gitlab/actions/discussions/winbar.lua

```diff
@@ -143,6 +143,60 @@ local add_drafts_and_resolvable = function(
   return base_title
 end
 
+-- Returns true if the input string (after removing highlighting strings) is wider than the
+-- discussion split.
+---@return boolean
+local too_long = function(str)
+  local d = require("gitlab.actions.discussions")
+  return #str:gsub("%%#[^#]+#", "") > vim.fn.winwidth(d.split.winid)
+end
+
+-- Returns the input winbar string shortened to fit into the windo width.
+---@return string
+local adapt_to_winwidth = function(str)
+  if too_long(str) then
```

Comment:
The solution is not particularly elegant, but unfortunately, Lua tables do not keep order of insertion, so it's not possible to do simply something like this if we want the `winbar` to adapt in a predictable way:
```lua
local shortenings = {
  ["Inline "] = "".
  [" Mode"] = "",
}
for pattern, replacement in pairs(shortenings) do
  if too_long(str) then
    str:gsub(pattern, replacement)
  end
end
```